### ISSUE TITLE
gradient function for tf.where

### ISSUE THREAD

**Issue by julianoks** [tensorflow/tfjs#338]
While trying to use tf.where, I found that it does not have a gradient function, and thus cannot be used in training. It would be rather trivial to make the gradient function, which would look something like:
```
{a: dy.mul(condition),
b: dy.mul(condition.logicalNot())}
```
Is this intentional? Will the gradient function be included in future releases? If not, is there a way to register my own gradient function without rebuilding the whole package?

**Comment by tafsiri** [comment]
I believe a gradient implementation for `tf.where` would be quite welcome. We have a list of desired gradients here https://github.com/tensorflow/tfjs/issues/108, its mostly an issue of bandwidth to implement them.

For registering your own gradient we have https://js.tensorflow.org/api/latest/#customGrad which would let you create a custom op that does calls tf.where on the forward pass and a custom function on the backwards pass.

**Comment by nsthorat** [comment]
This is fixed now!
